### PR TITLE
Feature scaling on training set when computing feature importances

### PR DESCRIPTION
**Pull request by Lilly-May**
**Description of changes**

As @eroell kindly pointed out, features in `ep.tl.feature_importances` were previously scaled based on the data from both the training and test sets. I changed this so that the scaler is now fitted exclusively on the training data and then used to transform the data of both the training and test sets.

### CODE REVIEW

**Review by Zethson — APPROVED** [review]
LGTM :) Thanks!

File: ehrapy/tools/feature_ranking/_feature_importances.py [hunk 1]
```diff
@@ -120,22 +120,26 @@ def rank_features_supervised(
     input_data = data[input_features]
     labels = data[predicted_feature]
 
+    x_train, x_test, y_train, y_test = train_test_split(input_data, labels, test_size=test_split_size)
```
Comment (eroell) [inline ehrapy/tools/feature_ranking/_feature_importances.py:123]: sorry for hopping in - what do you think about adding a random_state here? :)

Comment (eroell) [inline ehrapy/tools/feature_ranking/_feature_importances.py:123]:
```suggestion
    x_train, x_test, y_train, y_test = train_test_split(input_data, labels, test_size=test_split_size, random_state=42)
    
```

Comment (Lilly-May) [reply ehrapy/tools/feature_ranking/_feature_importances.py:123]: Good idea, thanks!

**Pull request merged**